lib/tasks/multi_region_kms.rake

```diff
@@ -0,0 +1,89 @@
+namespace :multi_region_kms do
+  desc 'Confirm that the multi-region KMS inner-layers are the same for both ciphertexts'
+  task check_inner_layer: :environment do |_task, _args|
+    ActiveRecord::Base.connection.execute('SET statement_timeout = 60000')
+
+    sample_password_users =
+      User.where.not(encrypted_password_digest_multi_region: nil).limit(10000).all
+    sample_personal_key_users =
+      User.where.not(encrypted_recovery_code_digest_multi_region: nil).limit(10000).all
+    sample_profiles = Profile.where.not(encrypted_pii_multi_region: nil).limit(10000).all
+
+    kms_client = Encryption::KmsClient.new
+
+    mismatched_records = []
+
+    sample_password_users.each do |user|
+      kms_context = {
+        'context' => 'password-digest',
+        'user_uuid' => user.uuid,
+      }
+
+      mr_inner_layer = kms_client.decrypt(
+        JSON.parse(user.encrypted_password_digest_multi_region)['encrypted_password'], kms_context
+      )
+      sr_inner_layer = kms_client.decrypt(
+        JSON.parse(user.encrypted_password_digest)['encrypted_password'], kms_context
+      )
+
+      if mr_inner_layer != sr_inner_layer
+        warn "Mismatch identified: User##{user.id}"
+        mismatched_records.push(user)
+      end
+    end
+
+    sample_personal_key_users.each do |user|
+      kms_context = {
+        'context' => 'password-digest',
```

Comment:
Should this context refer to the personal key? I dug around in the code for a bit and couldn't confirm.

Reply:
The personal key and password both use the password verifier which uses the `password-digest` context.